students/eowyn/session04/revised_mailroom.py

```diff
@@ -0,0 +1,123 @@
+        arg_dict = {"1": thank_you,
+                    "2": print_report,
+                    "3": send_letters,
+                    "4": quit_code}
+        try:
+            arg_dict.get(answer)()
+        except TypeError:
+            continue
+
+
+def quit_code():
+    sys.exit()
+
+
+def thank_you():
+    ''' Update donor records and print thank you notes.'''
+    while True:
+        fullname = input("To send a thank you, "
+                         "Enter a donor name (new or existing)\n"
+                         "Enter all to list existing DONORS \n"
+                         "Enter menu to return to main menu \n >")
+        fullname = remove_inputquotes(fullname)
+        if fullname == 'menu':
+            break
+        if fullname.lower() == 'all':
+            print("All Donors:")
+            [print(x) for x in DONORS]
+        elif fullname in DONORS:
+            print("Existing Donor")
+            DONORS[fullname].append(float(input("Donation amount: ")))
+            print(generate_letter(fullname))
+        else:
+            print("New Donor")
+            # must be a list to append later
+            DONORS[fullname] = [float(input("Donation amount: "))]
+            print(generate_letter(fullname))
+
+
+def print_report():
+    ''' Pretty-print a table of donor statistics '''
+    while True:
+        choice = input("To generate a summary report, \n"
+                       "type run now.\n"
+                       "To return to the main menu,\n"
+                       "type menu now>")
+        choice = remove_inputquotes(choice)
+        if choice == 'menu':
+            break
+        else:
+            [name, total, number, ave] = [[], [], [], []]
+            for x in DONORS:  # loop over the keys which are donor names
+                name.append(x)
+                total.append(round(sum(DONORS[x])))
+                number.append(len(DONORS[x]))
+                ave.append(round(total[-1] / number[-1]))
+            report_data = list(zip(name, total, number, ave))
+            report_data = sorted(
+                report_data, key=lambda y: int(y[1]), reverse=True)
+            print_table(report_data)
+
+
+def send_letters():
+    ''' Send thank you notes to everyone in the DONORS dict'''
+    for donor in DONORS:
+        outfile = donor.replace(" ","_") + '.txt'
+        with open(outfile, 'w') as f:
+            f.write(generate_letter(donor))
+    print("Successfully saved letters for each donor.")
+
+
+def generate_letter(donor_name):
+    '''
+    Generate a formatted thank you note to a donor and return
+    the string of the thank you note for printing or writing.
+    '''
+    fs = "Thank you, {}, for your generosity and recent gift of ${}."
+    return fs.format(donor_name, DONORS[donor_name][-1])
+
+
+def remove_inputquotes(a_string):
+    '''check for and remove anxillary quotes'''
+    if a_string[0] == a_string[-1] == '"' or a_string[0] == a_string[-1] == "'":
```

Comment:
python's nifty string methods make this MUCH easier:

`a_string.replace('"', '')`

or:

`a_string.strip('"')`